main.nf

```diff
@@ -13,72 +13,163 @@ nextflow.enable.dsl = 2
-    log.info logo + paramsHelp(command) + citation + NfcoreTemplate.dashedLine(params.monochrome_logs)
-    System.exit(0)
-}
-
-// Validate input parameters
-if (params.validate_params) {
-    validateParameters()
-}
-
-WorkflowMain.initialise(workflow, params, log, args)
+params.colabfold_alphafold2_params      = getColabfoldAlphafold2Params()
+params.colabfold_alphafold2_params_path = getColabfoldAlphafold2ParamsPath()
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     NAMED WORKFLOW FOR PIPELINE
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-if (params.mode == "alphafold2") {
-    include { ALPHAFOLD2 } from './workflows/alphafold2'
-} else if (params.mode == "colabfold") {
-    include { COLABFOLD } from './workflows/colabfold'
-} else if (params.mode == "esmfold") {
-    include { ESMFOLD } from './workflows/esmfold'
-}
-
+//
+// WORKFLOW: Run main analysis pipeline
+//
 workflow NFCORE_PROTEINFOLD {
+
+    main:
+    ch_multiqc  = Channel.empty()
+    ch_versions = Channel.empty()
+
     //
     // WORKFLOW: Run alphafold2
     //
     if(params.mode == "alphafold2") {
-        ALPHAFOLD2 ()
+        //
+        // SUBWORKFLOW: Prepare Alphafold2 DBs
+        //
+        PREPARE_ALPHAFOLD2_DBS (
+            params.alphafold2_db,
+            params.full_dbs,
+            params.bfd_path,
+            params.small_bfd_path,
+            params.alphafold2_params_path,
+            params.mgnify_path,
+            params.pdb70_path,
+            params.pdb_mmcif_path,
+            params.uniref30_alphafold2_path,
+            params.uniref90_path,
+            params.pdb_seqres_path,
+            params.uniprot_path,
+            params.uniprot_sprot,
+            params.uniprot_trembl
+        )
+        ch_versions = ch_versions.mix(PREPARE_ALPHAFOLD2_DBS.out.versions)
+
+        //
+        // WORKFLOW: Run nf-core/alphafold2 workflow
+        //
+        ALPHAFOLD2 (
+            ch_versions,
+            params.full_dbs,
+            params.alphafold2_mode,
+            params.alphafold2_model_preset,
+            PREPARE_ALPHAFOLD2_DBS.out.params,
+            PREPARE_ALPHAFOLD2_DBS.out.bfd.ifEmpty([]),
+            PREPARE_ALPHAFOLD2_DBS.out.small_bfd.ifEmpty([]),
+            PREPARE_ALPHAFOLD2_DBS.out.mgnify,
+            PREPARE_ALPHAFOLD2_DBS.out.pdb70,
+            PREPARE_ALPHAFOLD2_DBS.out.pdb_mmcif,
+            PREPARE_ALPHAFOLD2_DBS.out.uniref30,
+            PREPARE_ALPHAFOLD2_DBS.out.uniref90,
+            PREPARE_ALPHAFOLD2_DBS.out.pdb_seqres,
+            PREPARE_ALPHAFOLD2_DBS.out.uniprot
+        )
+        ch_multiqc  = ALPHAFOLD2.out.multiqc_report
+        ch_versions = ch_versions.mix(ALPHAFOLD2.out.versions)
     }
 
     //
     // WORKFLOW: Run colabfold
     //
     else if(params.mode == "colabfold") {
```

Comment:
If we just leave it if instead of else if, we can have the option Cedric wanted, of running multiple tools in the same run. 
What do you think?

Reply:
We would have to mix the channels of versions and multiqc too, i am not sure how tricky it would be for multiqc

Reply:
This will be for a different PR, will open an issue though. Also, I am starting to think that we could do it for a 1.2 version and leave this one as it is.